Sources/Rendering/Core/Mapper/CoincidentTopologyHelper.js

```diff
@@ -5,8 +5,26 @@ import macro from 'vtk.js/Sources/macros';
 function addCoincidentTopologyMethods(publicAPI, model, nameList) {
   nameList.forEach((item) => {
     publicAPI[`get${item.method}`] = () => model[item.key];
-    publicAPI[`set${item.method}`] = (factor, offset) => {
-      model[item.key] = { factor, offset };
+    publicAPI[`set${item.method}`] = function setCoincidentTopologyParams(
+      factor,
+      offset
+    ) {
+      let newFactor;
+      let newOffset;
+      if (arguments.length === 1) {
+        // first param is an object containing the properties
+        ({ factor: newFactor, offset: newOffset } = factor);
+      } else {
+        newFactor = factor;
+        newOffset = offset;
+      }
+
+      const changed =
+        model[item.key]?.factor !== newFactor ||
+        model[item.key]?.offset !== newOffset;
+
+      model[item.key] = { factor: newFactor, offset: newOffset };
+      return changed;
     };
```

Comment:
It looks like it would be easier to do that:
```
macro.set(publicAPI, model, [{ type: 'object', name: item.method}]);
```
This would only support calls such as: `setCoincidentTopologyParameters({factor:1, offset:1});` (already the case in `WidgetRepresentation`)

I guess we could also extend  `vtkMacros.findSetter()` so that we would do: 
```
macro.set(publicAPI, model, [{ type: 'object', name: item.method, params:['factor', 'offset']}]);
```

and that would allow `setCoincidentTopologyParameters(1, 1);` in addition to  `setCoincidentTopologyParameters({factor:1, offset:1});`

Reply:
Ah, thanks for reminding me of that setter format. I agree that having a `params` field would be useful, and I'm satisfied with the way you've structured it so I'll add that in.